aws/resource_aws_lex_slot_type_test.go

```diff
@@ -0,0 +1,402 @@
+	rName := "aws_lex_slot_type.test"
+	testSlotTypeID := "test_slot_type_" + acctest.RandStringFromCharSet(8, acctest.CharSetAlpha)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAwsLexSlotTypeNotExists(testSlotTypeID, LexSlotTypeVersionLatest),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAwsLexSlotTypeConfig_basic(testSlotTypeID),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccCheckAwsLexSlotTypeExists(testSlotTypeID, &v),
+					resource.TestCheckResourceAttr(rName, "value_selection_strategy", lexmodelbuildingservice.SlotValueSelectionStrategyOriginalValue),
+				),
+			},
+			{
+				ResourceName:            rName,
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"create_version"},
+			},
+			{
+				Config: testAccAwsLexSlotTypeConfig_valueSelectionStrategy(testSlotTypeID),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccCheckAwsLexSlotTypeExists(testSlotTypeID, &v),
+					resource.TestCheckResourceAttr(rName, "value_selection_strategy", lexmodelbuildingservice.SlotValueSelectionStrategyTopResolution),
+				),
+			},
+			{
+				ResourceName:            rName,
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"create_version"},
+			},
+		},
+	})
+}
+
+func TestAccAwsLexSlotType_disappears(t *testing.T) {
+	var v lexmodelbuildingservice.GetSlotTypeOutput
+	resourceName := "aws_lex_slot_type.test"
+	rName := "test_slot_type_" + acctest.RandStringFromCharSet(8, acctest.CharSetAlpha)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAwsLexSlotTypeNotExists(rName, LexSlotTypeVersionLatest),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAwsLexSlotTypeConfig_basic(rName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsLexSlotTypeExists(rName, &v),
+					testAccCheckResourceDisappears(testAccProvider, resourceAwsLexSlotType(), resourceName),
+				),
+				ExpectNonEmptyPlan: true,
+			},
+		},
+	})
+}
+
+func testAccCheckAwsLexSlotTypeExistsWithVersion(slotTypeName, slotTypeVersion string, output *lexmodelbuildingservice.GetSlotTypeOutput) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		var err error
+		conn := testAccProvider.Meta().(*AWSClient).lexmodelconn
+
+		output, err = conn.GetSlotType(&lexmodelbuildingservice.GetSlotTypeInput{
+			Name:    aws.String(slotTypeName),
+			Version: aws.String(slotTypeVersion),
+		})
+		if isAWSErr(err, lexmodelbuildingservice.ErrCodeNotFoundException, "") {
+			return fmt.Errorf("error slot type %s not found, %s", slotTypeName, err)
+		}
+		if err != nil {
+			return fmt.Errorf("error getting slot type %s: %s", slotTypeName, err)
+		}
+
+		return nil
+	}
+}
```

Comment:
The pattern we use for `Exists` functions is to pass the resource address, e.g. `aws_lex_slot_type.test` and then look up relevant information from the Terraform state to call the AWS API.

Something like the following. This also specifies which version isn't found in the error messages for completeness
```go
func testAccCheckAwsLexSlotTypeExistsWithVersion(n, slotTypeVersion string, res *lexmodelbuildingservice.GetSlotTypeOutput) resource.TestCheckFunc {
	return func(s *terraform.State) error {
		rs, ok := s.RootModule().Resources[n]
		if !ok {
			return fmt.Errorf("Not found: %s", n)
		}

		if rs.Primary.ID == "" {
			return fmt.Errorf("No Lex slot type ID is set")
		}

		conn := testAccProvider.Meta().(*AWSClient).lexmodelconn

		output, err := conn.GetSlotType(&lexmodelbuildingservice.GetSlotTypeInput{
			Name:    aws.String(rs.Primary.ID),
			Version: aws.String(slotTypeVersion),
		})
		if isAWSErr(err, lexmodelbuildingservice.ErrCodeNotFoundException, "") {
			return fmt.Errorf("error slot type %q version %s not found", rs.Primary.ID, slotTypeVersion)
		}
		if err != nil {
			return fmt.Errorf("error getting slot type %q version %s: %w", rs.Primary.ID, slotTypeVersion, err)
		}

		*res = *output

		return nil
	}
}
```